source/gui/settingsDialogs.py

```diff
@@ -478,6 +476,15 @@ def _getCategoryPanel(self, catId):
 					 "MultiCategorySettingsDialog.MIN_SIZE"
 					).format(cls, panel.Size[0])
 				)
+			panel.SetLabel("{} Settings".format(panel.title))
+			import oleacc
+			panel.server = nvdaControls.AccPropertyOverride(
+				panel,
+				propertyAnnotations={
+					oleacc.PROPID_ACC_ROLE: oleacc.ROLE_SYSTEM_PROPERTYPAGE,  # change the role from pane to property page
+					oleacc.PROPID_ACC_DESCRIPTION: panel.panelDescription,  # set a description

```

Comment:
What I've just noticed is that the NVDAObjects.behaviors.Dialog overlay is also applied to property pages, so it looks like you can safely remove the accDescription override for panels like this and only override the accRole. I have commented out this line and it did not result in different behavior.

Reply:
Hmm, yeah, you are right. The hard question is what to do about the dialog description. Either we need a way to stop the getDialogText on these functions, or a way to override the dialog title.

Reply:
What part of the dialog description do you mean in this case?

I really like the setup where the config profile is pronounced as part of the dialog description.

How about doing something like this:

1. Only have NVDA Settings in the dialog title.
2. Add a statictext with "editing {profilename}". Then, add the categories list and all other controls. This will make the profile name the first child on the dialog, which will be automatically announced using getDialogText. You no longer have to override accDescription for the dialog
3. When the current profile name changes, update the particular static text with the new profile name. Do something like this on the dialog:
`winUser.user32.NotifyWinEvent(winUser.EVENT_OBJECT_DESCRIPTIONCHANGE, settingsDialog.Handle,winUser.OBJID_CLIENT,winUser.CHILDID_SELF)`
    On the appModule, the only thing you need to do is override event_descriptionChange on the dialog to speak even when the dialog isn't the focus object.

This has the advantage that the visual behavior and non visually perceived behavior match much more than before.

Reply:
The difficult part with what you are proposing is finding a way to make the profile text look right within the dialog. It's not obvious how to achieve that.

> What part of the dialog description do you mean in this case?

I meant, after alt tab to the dialog. NVDA walks through to the focused control, speaking all static text until it gets there.  I now realise that as long as `PROPID_ACC_DESCRIPTION` is set to something other than white space this will not happen.

I like the idea of using the accessible description for the dialog to announce the profile name. Next I'm going to look into why NVDA does not seem to be getting the `PROPID_ACC_NAME` for the dialog. Inspect.exe shows it, but its not in NVDA's dev info.

Reply:
> The difficult part with what you are proposing is finding a way to make the profile text look right within the dialog. It's not obvious how to achieve that.

Isn't it possible to position it centered just below the window title?

Reply:
It's possible, I'll have to play with it to see how it looks. Because it's not editable it might look a little strange. I have been thinking it makes sense to be able to edit profiles that are not actually active, so maybe it should be a control at some stage.